feat: add extra attributes to device_tracker

andreasbrett opened the pull request:
Add 4 extra attributes to device tracker:

- speed
- course
- pdop (see https://en.wikipedia.org/wiki/Dilution_of_precision_(navigation))
- ~~confidence (GPS signal confidence)~~

Speed and course should be quite valuable since it allows HA to determine the direction of travel and speed. I'm not perfectly sure these are used for the [Proximity](https://www.home-assistant.io/integrations/proximity/) integration or if that is derived from the GPS location history. Since the HA mobile app is providing both attributes though and proximity sensors work very well with them, we should pass those attributes on from the Weenect API to HA.

PDOP and confidence on the other hand are for advanced users that want to dig deeper into GSM signal details. Basically they provide more detail on the GPS accuracy (when "number of GPS satellites" just isn't enough). I don't think it hurts to expose them via extra attributes. I didn't see them important enough to make dedicated sensors out of them though.

eifinger commented:
I saw you picked up on the naming scheme. I went for https://www.conventionalcommits.org/en/v1.0.0/ in this repo. In newer repos I do like https://github.com/release-drafter/release-drafter more but its a personal choice.

eifinger reviewed (changes requested):
Thank you for your contribution!

Can you please add those new attributes to the tests?

andreasbrett commented:
Added tests. While testing I could not get a confidence value in the API responses, so I removed it from this PR for now. I thought I had seen confidence values from the responses though. Might be my sickness atm though. If I pick up `confidence` in the near future I will raise another PR.
